src/main/drivers/rcdevice.c

```diff
@@ -0,0 +1,1143 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "common/crc.h"
+#include "common/maths.h"
+#include "drivers/time.h"
+#include "rcdevice.h"
+
+#include "config/feature.h"
+#include "fc/config.h"
+
+#include "io/serial.h"
+
+#include "io/rcdevice_cam.h"
+
+#define SAFE_FREE(x)                                                           \
+    {                                                                          \
+        if (x) {                                                               \
+            free(x);                                                           \
+            (x) = NULL;                                                        \
+        }                                                                      \
+    }
+
+typedef enum {
+    RCDP_SETTING_PARSE_WAITING_ID,
+    RCDP_SETTING_PARSE_WAITING_NAME,
+    RCDP_SETTING_PARSE_WAITING_VALUE,
+} runcamDeviceSettingParseStep_e;
+
+// the crc calc function for rcsplit 1.0 and 1.1, this function will
+// deprecate in feature 
+static uint8_t crc_high_first(uint8_t *ptr, uint8_t len)
+{
+    uint8_t i;
+    uint8_t crc=0x00;
+    while (len--) {
+        crc ^= *ptr++;
+        for (i=8; i>0; --i) {
```

Comment:
Declare variables at first point of usage, so use `for (int i = 8; i > 0; --i) {`